extra/import_data.py

```diff
@@ -24,19 +24,54 @@
 import pymongo
 import warnings
 import progressbar
+import argparse 
+import sys
 from openpyxl import load_workbook
-
+from argparse import RawTextHelpFormatter
+
+def argparser():
+    # Override of the error function in ArgumentParser from argparse
+    # To display the original error message + help page in the terminal, when the wrong or no arguments are given
+    class DefaultHelpParser(argparse.ArgumentParser):
+        def error(self, message):
+            sys.stderr.write('\nerror: %s\n' % message)
+            self.print_help()
+            sys.exit(1)  # exit program
+
+    # Override of adding an extra action class in Action from argparse
+    # To add an extra action that can check for file extensions and display the help page if wrong file extension
+    def CheckExt(choices):
+        class Act(argparse.Action):
+            def __call__(self, parser, namespace, fname, option_string=None):
+                for f in fname:
+                    ext = os.path.splitext(f)[1][1:]
+                    if ext not in choices:
+                        option_string = '({})'.format(option_string) if option_string else ''
+                        parser.error("file doesn't end with {}{}".format(choices, option_string))  # display error message
+                    else:
+                        setattr(namespace, self.dest, fname)
+        return Act
+
+    # required arguments and optional arguments for the terminal. A help page is automatically generated.
+    parser = DefaultHelpParser(
+        description="VUSualizer import data:\nImports data from 'O:\AMG_bijlagen_WES_uitslagen' after transferring this data into\nthe Dwergeik UMCU server. An additional script will notice new files in this folder\nand use these as a parameter for this script. This will update the local MongoDB.",
```

Comment:
Je kan lange string verdelen met `()`, misschien wel netjes omdat je ook `\n` erin hebt zitten?
```python
parser = DefaultHelpParser(
    description=(
        "VUSualizer import data:\n"
        "Imports data from 'O:\AMG_bijlagen_WES_uitslagen' after transferring this data into\n"
        "the Dwergeik UMCU server. An additional script will notice new files in this folder\n"
        "and use these as a parameter for this script. This will update the local MongoDB."
    ),
    formatter_class=RawTextHelpFormatter
)

```

Reply:
weer iets geleerd. bedankt :)